### PR TITLE
Adding histograms 

### PR DESCRIPTION
**Pull request by juandlt-csu**
This PR is progress on #41.

- `generate_general_histogram` : takes input of a df and its index in a list (all_data_flagged) to generate histograms of all the data.
   - The function does not pull the histogram data from the ggplot object, but that information can be retrieved. 
- `generate_seasonal_histogram` : takes input of a df and arranges the histograms by season.

This should help to determine the ranges for the site parameter combos.

### CODE REVIEW

File: src/qaqc/01-organize_raw_data.Rmd [hunk 1]
```diff
@@ -700,14 +700,92 @@ weekly_daily_flag_plots <- function(site_arg, parameter_arg, flag_arg) {
...
+  }
+  
+  collated_hist <- ggarrange(plotlist = hist_list, nrow=2, ncol=2)
+  return(collated_hist)
+  
+}
 
-# Resulting list of objects with "archery" in the title
-archery_objects_df <- bind_rows(archery_objects)
+generate_seasonal_histogram(df=all_data_flagged[["archery-Temperature"]])
```
Comment (kathryn-willi) [inline src/qaqc/01-organize_raw_data.Rmd:790]: perhaps we start saving the figs produced in some `data` subfolder?

**Review by kathryn-willi — APPROVED** [review]

File: src/qaqc/01-organize_raw_data.Rmd [hunk 2]
```diff
@@ -703,11 +703,12 @@ weekly_daily_flag_plots("archery", "Temperature", "slope flag suspect")
 Histograms of all parameter/site combos
 ```{r}
 # generate basic histograms for dfs
-generate_histograms <- function(df, df_index) {
+generate_general_histogram <- function(df, df_index) {
   
   site_param <- toupper(sub("-", " ", df_index, fixed = TRUE))
   title <- paste("Histogram of ", site_param)
```
Comment (SamStruthers) [inline src/qaqc/01-organize_raw_data.Rmd:709]: can we add a (n= number of measurements)?

File: src/qaqc/01-organize_raw_data.Rmd [hunk 3]
```diff
@@ -720,53 +721,71 @@ generate_histograms <- function(df, df_index) {
...
-  df <- all_data_flagged[["archery-Temperature"]] %>%
-    filter(month %in% seasons[[i]]) 
+generate_seasonal_histogram <- function(df) {
   
-  if (all(is.finite(df$mean))) {
-    
-    minimum <- floor(min(df$mean, na.rm = TRUE))
-    maximum <- ceiling(max(df$mean, na.rm = TRUE))
+  winter <- c(12,1,2)
+  spring <- c(3,4,5)
```
Comment (SamStruthers) [inline src/qaqc/01-organize_raw_data.Rmd:747]: I wonder if these seasons is the best way to think about it? Might be better to look at more hydrology related seasons
Snowmelt: 4,5, 6
Monsoon: 7, 8
Fall Baseflow: 9, 10, 11
Winter baseflow: 12, 1, 2, 3

Comment (kathryn-willi) [reply src/qaqc/01-organize_raw_data.Rmd:747]: I agree

File: src/qaqc/01-organize_raw_data.Rmd (hunk 2 above)
Comment (juandlt-csu) [reply src/qaqc/01-organize_raw_data.Rmd:709]: I don't see why not, I'll add that to the histogram issue.

File: src/qaqc/01-organize_raw_data.Rmd (hunk 3 above)
Comment (juandlt-csu) [reply src/qaqc/01-organize_raw_data.Rmd:747]: I also agree, I will add this to the histogram issue.

File: src/qaqc/01-organize_raw_data.Rmd (hunk 1 above)
Comment (juandlt-csu) [reply src/qaqc/01-organize_raw_data.Rmd:790]: This makes sense to me.

File: src/qaqc/01-organize_raw_data.Rmd [hunk 4]
```diff
@@ -700,14 +700,92 @@ weekly_daily_flag_plots <- function(site_arg, parameter_arg, flag_arg) {
...
+
+
+
+# making histograms for temperatures
+## pulling all the temperature dfs from the all_data list
+temperature_hist_list <- all_data_flagged[grep("Temperature", names(all_data_summary_list), ignore.case = TRUE)]
+
+## use imap to use the index in the list for the plot titles
+## (not loving imap, if there is a better way to do this please change this)
+temperature_hist_plots <- imap(temperature_hist_list, ~generate_general_histogram(.x, .y))
```
[the remark this answers is not in the document]
Comment (juandlt-csu) [reply src/qaqc/01-organize_raw_data.Rmd:732]: This will have to do for now then..

**Pull request merged**